dask_geopandas/clip.py

```diff
@@ -0,0 +1,38 @@
+import numpy as np
+import geopandas
+
+from dask.highlevelgraph import HighLevelGraph
+from dask.utils import derived_from
+from dask.base import tokenize
+
+from .core import GeoDataFrame
+
+
+@derived_from(geopandas.tools)
+def clip(gdf, mask, keep_geom_type=False):
+    if gdf.spatial_partitions is None:
+        return gdf.map_partitions(
+            lambda partition: geopandas.clip(
+                gdf=partition, mask=mask, keep_geom_type=keep_geom_type
+            ),
+            token="clip",
+            meta=gdf._meta,
+        )
+
+    new_spatial_partitions = geopandas.clip(
+        gdf=gdf.spatial_partitions.to_frame("geometry"),
+        mask=mask,
```

Comment:
Can you add the comment about why `keep_geom_type` is always set to False for the spatial partitions here in the code as well (the one you posted in the PR top post for the commit message).  
I would maybe also add `keep_geom_type=False` in the line above to be explicit in the code.